hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,82 +10,95 @@ This module conforms to the [Comportable guideline for peripheral functionality.
-  - The internal health tests have detected an adaptive proportion test fail.
+  - The internal health tests have detected a test fail.
   - An internal FIFO error has occurred.
 - Two health checks that are defined by SP 800-90B are performed by this revision:  Repetition Count and Adaptive Proportion tests.
-- No alerts are supported by this revision.
+- Two additional health are performed by this revision:  Bucket and Markov tests.
+- A health fail alert is supported by this revision.
 
 
 ## Description
 
 This IP block provides an entropy source that is part of a larger solution to generate true random numbers (a TRNG mechanism).
 The general solution for TRNG relies on either hardware or firmware to implement this function, depending which functions are enabled by firmware or e-fuses.
-For purposes of reference in this document, the NIST term "noise" will be synonymous with the term "entropy".
-Likewise, no conditioning function will be performed in this block.
+A conditioning function will be performed in this block.
 For the generation of entropy, this block provides the entropy bits needed for the entropy source solution as required by the RNG mechanism.
 
 At a high-level, this ENTROPY_SRC block, when enabled, will continuously collect entropy bits from the entropy source into a FIFO that can be read from the TLUL bus, or sent out through a hardware interface.
-The rng source will generate a relative slow set of bit streams, in the tens of kilohertz range.
+The RNG source will generate a relatively slow set of bit streams, in the tens of kilohertz range.
 The digital source can be programmed to be faster, or much slower if desired.
-A digital source rate of flexiable speed might be useful in demonstrating how the downstream applications use the entropy.
-
-Once the threshold has been reached for collected entropy bits, the firmware has two options of notification: an interrupt, or polling.
-The `es_entropy_valid` bit will be set when the {{< regref "ES_THRESH" >}} register value is equal or less than the number of elements in the TLUL FIFO.
-Optionally, an interrupt may be a better use of CPU cycles.
+A digital source rate of flexible speed might be useful in demonstrating how the downstream applications use the entropy.
+
+Once a noise source has been selected through configuration, the raw noise data processing begins.
+This ENTROPY_SRC block will do health checks, collect and pack bits, and finally send them through a conditioning unit. 
+There are two modes in which entropy bits are delivered, bypass and FIPS/CC mode. 
+Bypass mode will deliver faster bits for chip boot usage, and the bits are not required to be FIPS/CC compliant entropy.
+This means that the min-entropy value is a relatively lower and acceptable value for bypass mode.
+There are several test statistics that are kept in registers to provide diagnostic capabilities when examining real entropy test fails.
```

Comment:
what does "real" mean in this context?  Do you mean "observed"?

Reply:
Agreed.  "Real" is a slippery term here, and based on my conversations with @mwbranstad, my understanding is that these are not just meant to be consulted when there is a health check failure, but any time when doing diagnostics.
 So I changed it to say: "In addition to the brief, low-latency health checks, various long-term statistics are accumulated in registers for
in-depth analysis"